microprofile/WFLY-17510-remove-microprofile-opentracing.adoc

```diff
@@ -0,0 +1,100 @@
+= WFLY-17510: Remove MicroProfile OpenTracing
+:author:            Jason Lee
+:email:             jasondlee@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+WildFly has long offered MicroProfile OpenTracing as part of its overall MicroProfile support. Since its inclusion in WildFly, however, OpenTracing has merged with OpenCensus to form https://opentelemetry.io[OpenTelemetry]. Because of this merger, the OpenTracing project itself has been abandoned, and the MicroProfile Working Group has decided to remove OpenTracing from the platform specification, replacing it with MicroProfile Telemetry, which wraps and exposes https://opentelemetry.io[OpenTelemetry]. The effort under this JIRA will cover only the effort to remove MicroProfile OpenTracing (MicroProfile OpenTracing) from the server and migrate installations to MicroProfile Telemetry.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFLY-17510[WFLY-17510]
+
+=== Related Issues
+
+* https://issues.redhat.com/browse/WFLY-17156[WFLY-17156]
+* https://issues.redhat.com/browse/EAP7-2002[EAP7-2002]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+=== Testing By
+* [x] Engineering
+* [ ] QE
+
+=== Affected Projects or Components
+
+All the code changes will be confined to the MicroProfile OpenTracing extension and subsystem, which includes the Java code, Galleon modules/layers, etc.
+
+=== Other Interested Projects
+
+OpenTelemetry/MicroProfile Telemetry are involved but unchanged.
+
+=== Relevant Installation Types
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+* [ ] Managed domain
+* [x] OpenShift s2i
+* [x] Bootable jar
+
+== Requirements
+
+=== Hard Requirements
+
+MicroProfile OpenTracing will be migrated to an admin-only module
+
+A `MigrateOperation` will be added to remove the extension from existing server configs upon server startup (in admin-only mode). For systems with MicroProfile OpenTracing configured, OpenTelemetry will be configured using default values.
+
+=== Nice-to-Have Requirements
+
+=== Non-Requirements
+
+Converting MicroProfile OpenTracing config to OpenTelemetry config. The configuration options and default values vary enough that a meaningful conversion is not possible.
+
+== Backwards Compatibility
+
+This represents the removal of a subsystem, so any application the depends on these APIs will fail to deploy. Likewise, any installation that is expecting (Jaeger) tracing data to be exported will see a behavior change that may or may not impact any external monitoring solutions.
+
+All other subsystems should be unaffected by this removal.
+
+=== Default Configuration
+
+The default configuration for MicroProfile configurations will be altered to remove the MicroProfile OpenTracing subsystem. Likewise, the relevant Galleon layers will also have the extension removed from their definitions.
```

Comment:
Is it Galleon layers or feature packs? I am not sure though, just asking.

Reply:
I think, both actually. There are `layer-spec.xml` files, as well as feature pack definitions. @bstansberry can chime in if I'm off base here. :)

Reply:
It's layers.

The feature pack still incorporates MP OT as a FP incorporates everything that can get provisioned, and MP OT can get provisioned. The galleon content used for the default configs and the layers are what drive whether MP OT shows up in the configuration files. That's what we want to change.

That said, the words "the relevant Galleon layers" should be replaced with an explicit list of the layers that are going to change.

Reply:
I'll update that text to include those layers.

Reply:
@bstansberry @fabiobrz Edited the text. Is that better?

Reply:
Yep, I think it solves Brian's concerns.